mapper/javabean/src/main/java/org/hibernate/search/mapper/javabean/loading/impl/JavaBeanSearchLoadingContext.java

```diff
@@ -47,18 +63,78 @@ public Object loaderKey(PojoRawTypeIdentifier<?> type) {
 	public <T> PojoLoader<T> createLoader(Set<PojoRawTypeIdentifier<? extends T>> expectedTypes) {
 		PojoRawTypeIdentifier<? extends T> type = expectedTypes.iterator().next();
 		PojoLoader<T> loader = (PojoLoader<T>) loaderByType.get( type );
+		if ( loader == null ) {
+			for ( Map.Entry<PojoRawTypeIdentifier<?>, PojoLoader<?>> entry : loaderByType.entrySet() ) {
+				if ( entry.getKey().javaClass().isAssignableFrom( type.javaClass() ) ) {
+					loader = (PojoLoader<T>) entry.getValue();
+					break;
+				}
+			}
+		}
 		if ( loader == null ) {
 			throw log.entityLoaderNotRegistered( type );
 		}
 		return loader;
 	}
 
-	public static final class Builder implements PojoLoadingContextBuilder<LoadingOptions>, LoadingOptions {
+	@Override
+	public Object indexingKey(PojoRawTypeIdentifier<?> type) {
+		return type;
+	}
+
+	@Override
+	public String entityName(PojoRawTypeIdentifier<?> entityType) {
+		LoadingTypeContext typeContext = typeContextProvider.indexedForExactType( entityType );
```

Comment:
LoadingTypeContext is a raw type here, it should have a type parameter.

Please also have a look at the other raw types detected by Sonar: https://sonarcloud.io/project/issues?id=org.hibernate.search%3Ahibernate-search-parent&open=AXhNbBcuevIw7MZGvMo-&pullRequest=2503&resolved=false&rules=java%3AS3740&types=CODE_SMELL

Reply:
Many of these types are unrelated to change. The same was true of the first version carried over. The problem was before. But of course, being in this fragment, I can improve it.